pkg/tinydtls/contrib/sock_dtls.c

```diff
@@ -84,19 +84,11 @@ static int _read(struct dtls_context_t *ctx, session_t *session, uint8_t *buf,
                  size_t len)
 {
     (void)session;
-    msg_t msg = { .type = DTLS_EVENT_READ };
     sock_dtls_t *sock = dtls_get_app_data(ctx);
 
     DEBUG("sock_dtls: decrypted message arrived\n");
-    if (sock->buflen < len && sock->buf) {
-        DEBUG("sock_dtls: not enough place on buffer for decrypted message\n");
-        msg.content.value = -ENOBUFS;
-    }
-    else {
-        memmove(sock->buf, buf, len);
```

Comment:
Removing the handling (copy) of the data and just saving the pointer and length looks a bit dangerous to me. As it is now it should work because `_get_buffer` will always be called before `dtls_handle_message` is called again. But I'm not sure this will always be the case if `handle message` and `_get_buffer` are decoupled in some async way.
This way the pointer will be used outside of it's intended scope (i.e. the `_read` callback). Also we can not guarantee (or should rely on) `dtls_handle_message` calling read only a single time, right? Am I missing something here?

Reply:
In tracking the calls of `dtls_handle_message`, I don't see how data can be lost with the current approach compared to the previous.

Reply:
> I don't see how data can be lost

What about the case when _read is called multiple times from dtls_handle_message?

Reply:
Then the current way doesn't work either. `sock->buf` is overwritten to the last record in either way. Even worse, the current approach suggests, there are multiple records (by having multiple messages in the `mbox`), but only the last data point is contained (as the data is not stored in the `mbox` but in `sock->buf` which is singleton).

Reply:
You are right, I thought in the previous solution the target buffer was updated before the next _read (memmove) can be called. Anyway, I think a solution for this (or a clarification that it really is unneeded) should be targeted at some point.